scripts/biomedical/VirtualMetabolicHuman/format_microbes.py

```diff
@@ -0,0 +1,149 @@
+@file_input: input .tsv from VMH with microbiome data
+@file_output: csv output file with enums and dcids generated
+'''
+import sys
+import pandas as pd
+import datacommons as dc
+
+
+def modify_org_name(df_microbes):
+    """
+    Modifies the organism name in the dataframe for an
+    efficient sparql query
+    Arg:
+        Dataframe with microbe names
+    Returns:
+        List with modified microbe names    
+    """
+    org_name = ['0'] * len(df_microbes['organism'])
+    for i in df_microbes.index:
+        l = df_microbes.loc[i, "organism"].split(' ')
+        org_name[i] = l[0] + " " + l[1]
+    return org_name
+
+
+def org_name_query():
+    """
+    Performs a sparql query to retrieve all microbes names
+    on datacommons and wraps the species_dcids in a list
+    Arg: None
+    Returns:
+        list with specie dcids
+    """
+    query_str = '''
+    SELECT ?node
+    WHERE {
+    ?node typeOf Species .
+    ?node organismTaxonomicKingdom ?kingdom .
+    ?kingdom name Bacteria .
+    }
+    '''
+    result = dc.query(query_str)
+    species_dcids = ['0'] * len(result)
+    for index in range(len(result)):
+        for key in result[index]:
+            species_dcids[index] = result[index][key]
+    return species_dcids
+
+
+def scientific_name_query(species_dcids):
+    """
+    Fetches the scientific names from the list
+    of dcids
+    Arg:
+        species_dcids = list of dcids
+    Returns:
+        df_species = df with specie dcids and names
+    """
+    df_species = pd.DataFrame()
+    df_species['species_dcid'] = species_dcids
+
+    df_species['species_name'] = df_species['species_dcid'].map(
+        dc.get_property_values(df_species['species_dcid'], 'scientificName'))
+    df_species = df_species.explode('species_name')
+    return df_species
+
+
+def generate_dcid(df_species, org_name, df_microbes):
+    """
+    Adds dcids to name matches and generates
+    new ones if they don't exist on dc
+    Arg:
+        df_species = df with species names 
+        and dcids
+        org_name = list of microbe names 
+        from dc query
+        df_microbes = original microbes df
+    Returns:
+        df_microbes = microbes df with dcid addeed
+    """
+    species_count = ['0'] * len(df_microbes['organism'])
+    for i in range(len(org_name)):
+        for j in df_species.index:
+            if (org_name[i] == df_species.loc[j, "species_name"]):
+                org_name[i] = df_species.loc[j, "species_dcid"]
+                species_count[i] = '1'
+
+    #map dcids and create new ones, in case they don't exist on dc
+    for j in range(len(species_count)):
+        if species_count[j] == '0':
+            p = org_name[j].split(' ')
+            org_name[j] = "bio/" + (p[0][0].upper() + p[0][1].upper() +
+                                    p[0][2].upper() + p[1][0].upper() +
+                                    p[1][1].upper())
+
+    df_microbes.insert(0, 'Id', org_name)
+    return df_microbes
+
+
+def main():
```

Comment:
Again, too much stored under main() for all scripts

Reply:
Done !